kcov: 34 -> 35

Ekleog opened the pull request:
As the maintainer currently in `meta.maintainers` appears to not have
touched the package since 2015, I've also added myself there.

cc @FlorentBecker 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ekleog committed:
kcov: 34 -> 35

As the maintainer currently in `meta.maintainers` appears to not have
touched the package since 2015, I've also added myself there.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=bef6642a-85c2-4158-8b38-1ddc55566d9f)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35
shrinking /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35/bin/kcov
shrinking /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35/bin/kcov-system-daemon
gzipping man pages under /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35/share/man/
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35/bin
patching script interpreter paths in /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35
checking for references to /build in /nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35...
/nix/store/ci4rkshn97mnd911d3qi1p9gx53l7pzw-kcov-35
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=0342d4c4-c880-47a9-b125-1bed464b92e1)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
                                                ^~~~~~~~~~~
                                                ADJ_STATUS
make[2]: *** [src/CMakeFiles/kcov.dir/build.make:708: src/CMakeFiles/kcov.dir/engines/ptrace.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:244: src/CMakeFiles/kcov-system-daemon.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:207: src/CMakeFiles/kcov.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/n5p0326sqza7r9y9izwcav7k2i7syvg4-kcov-35.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/n5p0326sqza7r9y9izwcav7k2i7syvg4-kcov-35.drv' failed
```
</p></details>

Ekleog committed:
kcov: fix build on aarch64

Apparently, without this patch `NT_PRSTATUS` is not found. So the patch
adds the include apparently necessary. `NT_PRSTATUS` is also defined in
`<linux/ptrace.h>`, which would likely have been a better name, were it
not in the `linux/` directory, which is a priori not stable.

The need to do that is kind of weird (the change was introduced in [1],
and fedora apparently didn't need this additional import), but I'll try
to upstream it.

[1] https://github.com/SimonKagstrom/kcov/pull/239

Mic92 commented on pkgs/development/tools/analysis/kcov/default.nix:
```diff
@@ -2,20 +2,22 @@
 
   buildInputs = [ zlib curl elfutils python libiberty libopcodes ];
 
+  patches = [ ./aarch64_nt_prstatus.patch ];
```
Look useful. Is this pull request included in the next upstream version or does it come from a pull request?

Ekleog replied:
Just opened https://github.com/SimonKagstrom/kcov/pull/243 :)

Mic92 commented:
@GrahamcOfBorg build kcov

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=efa6f4da-2f52-4847-a339-9e838745d8f7)

The following builds were skipped because they don't evaluate on x86_64-darwin: kcov

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=e758bcc0-f993-4026-9415-e1c66b32cede)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35
shrinking /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35/bin/kcov
shrinking /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35/bin/kcov-system-daemon
gzipping man pages under /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35/share/man/
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35/bin
patching script interpreter paths in /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35
checking for references to /build in /nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35...
/nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=49ea1849-53ca-4e37-bfed-177fb427ce7d)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35
shrinking /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35/bin/kcov-system-daemon
shrinking /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35/bin/kcov
gzipping man pages under /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35/share/man/
strip is /nix/store/j7d4mr0ikv974ig7yzhknpsq288js4bs-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35/bin
patching script interpreter paths in /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35
checking for references to /build in /nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35...
/nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=9884ff3f-89ab-4f06-a231-c2faf6636b44)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/wh0x01sxfbm64calqzi0bnzm6hw3icrz-kcov-35
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39624&attempt_id=87031929-ea66-43ad-8d5e-b1ee6e140242)

Attempted: kcov

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/7rcfrlhx5pnq2z9jd5x5wb09mxnh2f7q-kcov-35
```
</p></details>
